C++ error in changed handler: expected ‘template’ keyword before dependent template name

### Bug Description

The Slint code below generates the following C++ code:

```cpp
auto AppWindow::user_init () -> void{
    [[maybe_unused]] auto self = this;
    self->change_tracker0.init(self, [](auto self) { return self->root_1_value.get(); }, []([[maybe_unused]] auto self, auto) { self->globals->window().window_handle().close_popup(self->popup_id_0); self->popup_id_0 = self->globals->window().window_handle().show_popup<Component_popup_2>(&*(self), [=](auto self) { return [&](const auto &a_0, const auto &a_1){ slint::private_api::LogicalPosition o{}; o.x = a_0; o.y = a_1; return o; }(self->popup_2_x.get(), self->popup_2_y.get()); }, slint::cbindgen_private::PopupClosePolicy::CloseOnClick, { self->self_weak.lock()->into_dyn(), self->tree_index }); });
}
```

Which leads to the compile error:

```
FAILED: CMakeFiles/my_application.dir/slint_generated_app-window_1.cpp.o 
/usr/bin/c++  -Ibuild -Ibuild/_deps/slint-build/generated_include -Ibuild/_deps/slint-src/api/cpp/include -std=gnu++20 -MD -MT CMakeFiles/my_application.dir/slint_generated_app-window_1.cpp.o -MF CMakeFiles/my_application.dir/slint_generated_app-window_1.cpp.o.d -o CMakeFiles/my_application.dir/slint_generated_app-window_1.cpp.o -c build/slint_generated_app-window_1.cpp
build/slint_generated_app-window_1.cpp: In lambda function:
build/slint_generated_app-window_1.cpp:198:259: warning: expected ‘template’ keyword before dependent template name [-Wmissing-template-keyword]
  198 |     self->change_tracker0.init(self, [](auto self) { return self->root_1_value.get(); }, []([[maybe_unused]] auto self, auto) { self->globals->window().window_handle().close_popup(self->popup_id_0); self->popup_id_0 = self->globals->window().window_handle().show_popup<Component_popup_2>(&*(self), [=](auto self) { return [&](const auto &a_0, const auto &a_1){ slint::private_api::LogicalPosition o{}; o.x = a_0; o.y = a_1; return o; }(self->popup_2_x.get(), self->popup_2_y.get()); }, slint::cbindgen_private::PopupClosePolicy::CloseOnClick, { self->self_weak.lock()->into_dyn(), self->tree_index }); });
      |                                                                                                                                                                                                                                                                   ^~~~~~~~~~
      |                                                                                                                                                                                                                                                                   template
build/slint_generated_app-window_1.cpp:198:287: error: expected primary-expression before ‘>’ token
  198 |     self->change_tracker0.init(self, [](auto self) { return self->root_1_value.get(); }, []([[maybe_unused]] auto self, auto) { self->globals->window().window_handle().close_popup(self->popup_id_0); self->popup_id_0 = self->globals->window().window_handle().show_popup<Component_popup_2>(&*(self), [=](auto self) { return [&](const auto &a_0, const auto &a_1){ slint::private_api::LogicalPosition o{}; o.x = a_0; o.y = a_1; return o; }(self->popup_2_x.get(), self->popup_2_y.get()); }, slint::cbindgen_private::PopupClosePolicy::CloseOnClick, { self->self_weak.lock()->into_dyn(), self->tree_index }); });
      |                                                                                                                                                                                                                                                                                               ^
build/slint_generated_app-window_1.cpp:198:544: error: expected primary-expression before ‘{’ token
  198 |     self->change_tracker0.init(self, [](auto self) { return self->root_1_value.get(); }, []([[maybe_unused]] auto self, auto) { self->globals->window().window_handle().close_popup(self->popup_id_0); self->popup_id_0 = self->globals->window().window_handle().show_popup<Component_popup_2>(&*(self), [=](auto self) { return [&](const auto &a_0, const auto &a_1){ slint::private_api::LogicalPosition o{}; o.x = a_0; o.y = a_1; return o; }(self->popup_2_x.get(), self->popup_2_y.get()); }, slint::cbindgen_private::PopupClosePolicy::CloseOnClick, { self->self_weak.lock()->into_dyn(), self->tree_index }); });
      |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                ^
build/slint_generated_app-window_1.cpp:198:543: error: expected ‘)’ before ‘{’ token
  198 |     self->change_tracker0.init(self, [](auto self) { return self->root_1_value.get(); }, []([[maybe_unused]] auto self, auto) { self->globals->window().window_handle().close_popup(self->popup_id_0); self->popup_id_0 = self->globals->window().window_handle().show_popup<Component_popup_2>(&*(self), [=](auto self) { return [&](const auto &a_0, const auto &a_1){ slint::private_api::LogicalPosition o{}; o.x = a_0; o.y = a_1; return o; }(self->popup_2_x.get(), self->popup_2_y.get()); }, slint::cbindgen_private::PopupClosePolicy::CloseOnClick, { self->self_weak.lock()->into_dyn(), self->tree_index }); });
      |                                                                                                                                                                                                                                                                                                ~                                                                                                                                                                                                                                                              ^~
      |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               )
build/slint_generated_app-window_1.cpp:198:600: error: expected ‘;’ before ‘)’ token
  198 |     self->change_tracker0.init(self, [](auto self) { return self->root_1_value.get(); }, []([[maybe_unused]] auto self, auto) { self->globals->window().window_handle().close_popup(self->popup_id_0); self->popup_id_0 = self->globals->window().window_handle().show_popup<Component_popup_2>(&*(self), [=](auto self) { return [&](const auto &a_0, const auto &a_1){ slint::private_api::LogicalPosition o{}; o.x = a_0; o.y = a_1; return o; }(self->popup_2_x.get(), self->popup_2_y.get()); }, slint::cbindgen_private::PopupClosePolicy::CloseOnClick, { self->self_weak.lock()->into_dyn(), self->tree_index }); });
      |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        ^
      |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        ;
ninja: build stopped: subcommand failed.
```

### Reproducible Code (if applicable)

```slint
export component AppWindow inherits Window {
    in-out property<string> value;

    changed value => {
        popup.show();
    }

    popup := PopupWindow {}
}
```

### Environment Details

- Slint Version: Reproducible with 1.11.0 and with current master
- Platform/OS: Linux
- Programming Language: C++
- Backend/Renderer: Qt


### Product Impact

_No response_

--- a/internal/compiler/generator/cpp.rs
+++ b/internal/compiler/generator/cpp.rs
@@ -3814,7 +3814,7 @@ fn compile_builtin_function_call(
                 let position = compile_expression(&popup.position.borrow(), &popup_ctx);
                 let close_policy = compile_expression(close_policy, ctx);
                 format!(
-                    "{window}.close_popup({component_access}->popup_id_{popup_index}); {component_access}->popup_id_{popup_index} = {window}.show_popup<{popup_window_id}>(&*({component_access}), [=](auto self) {{ return {position}; }}, {close_policy}, {{ {parent_component} }})"
+                    "{window}.close_popup({component_access}->popup_id_{popup_index}); {component_access}->popup_id_{popup_index} = {window}.template show_popup<{popup_window_id}>(&*({component_access}), [=](auto self) {{ return {position}; }}, {close_policy}, {{ {parent_component} }})"
                 )
             } else {
                 panic!("internal error: invalid args to ShowPopupWindow {arguments:?}")
@@ -3901,7 +3901,7 @@ fn compile_builtin_function_call(
             };
             format!(r"
                 {window}.close_popup({context_menu}.popup_id);
-                {context_menu}.popup_id = {window}.show_popup_menu<{popup_id}>({globals}, {position}, {{ {context_menu_rc} }}, [self](auto popup_menu) {{
+                {context_menu}.popup_id = {window}.template show_popup_menu<{popup_id}>({globals}, {position}, {{ {context_menu_rc} }}, [self](auto popup_menu) {{
                     auto parent_weak = self->self_weak;
                     auto self_ = self;
                     {init}

--- a/tests/cases/issues/issue_8710_popup_show_from_changed.slint
+++ b/tests/cases/issues/issue_8710_popup_show_from_changed.slint
@@ -1,0 +1,15 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-2.0 OR LicenseRef-Slint-Software-3.0
+
+// C++ code generation used to produce error for this test case
+
+export component AppWindow inherits Window {
+    in-out property<string> value;
+
+    changed value => {
+        popup.show();
+    }
+
+    popup := PopupWindow {}
+}
+

